src/node_platform.cc

```diff
@@ -0,0 +1,201 @@
+#include "node_platform.h"
+
+namespace node {
+
+using v8::Isolate;
+using v8::Platform;
+using v8::Task;
+using v8::TracingController;
+
+static void FlushTasks(uv_async_t* handle) {
+  NodePlatform* platform = static_cast<NodePlatform*>(handle->data);
+  platform->FlushForegroundTasksInternal();
+}
+
+static void DeleteHandle(uv_handle_t* handle) {
+  delete handle;
+}
+
+static void BackgroundRunner(void* data) {
+  TaskQueue<Task>* background_tasks = static_cast<TaskQueue<Task>*>(data);
+  Task* task = NULL;
+  while ((task = background_tasks->BlockingPop())) {
```

Comment:
You could shorten this to `while (Task* task = background_tasks->BlockingPop()) {`.

Also, `nullptr`, not `NULL`.